inference-engine/src/gna_plugin/backend/make_pwl.cpp

```diff
@@ -583,6 +649,7 @@ void make_gna_pwl(const DnnActivation  fun,
         }
         default:
             gnalog() << "Unexpected function activation!\n";
-            std::cerr << "Unexpected function activation!\n";
+            THROW_GNA_EXCEPTION << "Unexpected function activation!" << fun;
     }
+    insert_extra_pwl_segments(gna_pwl, y_min, y_max);
```

Comment:
consider to limit this insertion to CNN2D

Reply:
The problem is that at this place in the code it's hard to tell what's the previous layer.

Reply:
It is quite simple to pass this information starting from:
`void GNAGraphCompiler::PWLPrimitive(InferenceEngine::CNNLayerPtr layer)` 
and `void GNAGraphCompiler::PowerPrimitive(InferenceEngine::CNNLayerPtr layer)` functions.
The previous layer can be deduced there based on the `dnnComponents` list.

And passed through `PwlDesignOpt` function -> `make_gna_pwl` function.

@sirzabek consider to add a ticket for that at least.

Reply:
I agree. Because the information about CNN2D was added in the task in a late stage and it may require some thorough testing, I decided to create a new task for it: 54920.